core/src/HIP/Kokkos_HIP_IsXnack.hpp

```diff
@@ -45,6 +45,16 @@ so we infer its presence is not necessary.
 bool xnack_environment_enabled();
 // Returns true iff we detect CONFIG_HMM_MIROR=y in /boot/config-$(uname -r).
 bool xnack_boot_config_has_hmm_mirror();
+// Returns true iff the architecture of the gpu supports accessing system
+// allocated memory
+constexpr bool gpu_arch_can_access_system_allocations() {
+#if defined(KOKKOS_ARCH_AMD_GFX908) || defined(Kokkos_ARCH_AMD_GFX90A) || \
+    defined(Kokkos_ARCH_AMD_GFX942) || defined(Kokkos_ARCH_AMD_GFX942_APU)
```

Comment:
1. Include what you use `#include <Kokkos_Macros.hpp>`
2. Typos `Kokkos_* -> KOKKOS_*`

Reply:
Also generally I dislike spreading the number of places we need to keep in sync with respect of ARCH macros.

Reply:
I am with you on this. I made the function fail if a new arch is added instead of a fallback.